libifupdown/dict.c

```diff
@@ -49,6 +50,31 @@ lif_dict_add(struct lif_dict *dict, const char *key, void *data)
 	return entry;
 }
 
+struct lif_dict_entry *
+lif_dict_add_once(struct lif_dict *dict, const char *key, void *data,
+                  int (*compar)(const void *, const void *))
```

Comment:
If you use a typedef here, we can simply typecast strcmp.

Reply:
I sticked to the signature given in `man 3 qsort` to be as generic as possible. Hence the simple wrapper function (which we could enrich with an `inline` statement).